### ISSUE TITLE
cursor.cursor('path') to immutable.List returns wrong cursor type

### ISSUE THREAD

**Issue by jeroencranendonk-wf** [facebook/immutable-js#318]
```js
var data = Immutable.fromJS({ a: [1, 2, 3] });
var cursor = Cursor.from(data);
var deepCursor = cursor.cursor('a');
```

deepCursor will be KeyedCursor, not IndexedCursor, due to the way `subCursor()` calls `makeCursor()` and how `makeCursor()` checks for optional arguments.

#297 fixes this issue.

**Comment by leebyron** [comment]
Thanks for the report (and the pull request) ((and your patience)).

I'll dig into this and make sure this is fixed and incorporate your pr soon.

**Comment by jeroencranendonk-wf** [comment]
Awesome! Thanks.

**Comment by Gozala** [comment]
I've being bitten by this as, well. I hope it will make into a release soon!

**Comment by phated** [comment]
Just ran into this, looking forward to the fix.

**Comment by jeffbski** [comment]
+1

**Comment by jeffbski** [comment]
I posted a slightly revised fix here #381 which uses argument length since undefined might be a legitimate value.

**Comment by wmertens** [comment]
:+1:
